GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -126,14 +141,43 @@ - (void)storeEvent:(GDTCOREvent *)event
                                                expirationDate:event.expirationDate
                                                     mappingID:event.mappingID];
     NSError *error;
-    GDTCOREncodeArchive(event, filePath, &error);
+    NSData *encodedEvent = GDTCOREncodeArchive(event, nil, &error);
+    if (error) {
+      completion(NO, error);
+      return;
+    }
+
+    // Check storage size limit before storing the event.
+    uint64_t resultingStorageSize = self.sizeCalculator.directoryContentSize + encodedEvent.length;
+    if (resultingStorageSize > kGDTCORFlatFileStorageSizeLimit) {
```

Comment:
Instead of dropping the new events, can we drop the old events? There is a good chance that the stale events sitting on the device can force the recent events to get dropped.

Reply:
This is current behaviour on Android and Clearcut SDK. DD attached to b/129492246 describes several alternatives which we are going to iterate on simultaneously on Firelog iOS and Android.